soc/espressif/esp32/Kconfig.soc

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_ESP32
+	bool
+	select SOC_FAMILY_ESPRESSIF_ESP32
+	help
+	  ESP32 Series
+
+# SoC with/without embedded flash
+config SOC_ESP32_D0WD_V3
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_D0WD_V3
+
+config SOC_ESP32_D0WDR2_V3
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_D0WDR2_V3
+
+config SOC_ESP32_U4WDH
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_U4WDH
+
+config SOC_ESP32_PICO_V3
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_PICO_V3
+
+config SOC_ESP32_PICO_V3_02
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_PICO_V3_02
+
+config SOC_ESP32_PICO_D4
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_PICO_D4
+
+# SiP with external flash / psram
+config SOC_ESP32_WROOM_DA_N4
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_DA_N4
+
+config SOC_ESP32_WROOM_DA_N8
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_DA_N8
+
+config SOC_ESP32_WROOM_DA_N16
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_DA_N16
+
+config SOC_ESP32_WROOM_32UE_N4
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_32UE_N4
+
+config SOC_ESP32_WROOM_32UE_N8
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_32UE_N8
+
+config SOC_ESP32_WROOM_32UE_N16
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROOM_32UE_N16
+
+config SOC_ESP32_WROVER_E_N4R2
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N4R2
+
+config SOC_ESP32_WROVER_E_N8R2
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N8R2
+
+config SOC_ESP32_WROVER_E_N16R2
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N16R2
+
+config SOC_ESP32_WROVER_E_N4R8
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N4R8
+
+config SOC_ESP32_WROVER_E_N8R8
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N8R8
+
+config SOC_ESP32_WROVER_E_N16R8
+	bool
+	select SOC_SERIES_ESP32
+	help
+	  ESP32_WROVER_E_N16R8
+
+config SOC_ESP32_PROCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for PROCPU (core 0).
+
+config SOC_ESP32_APPCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
```

Comment:
rename and move to `Kconfig`

Reply:
@nordicjm moving this symbols to Kconfig will fail in board level during compliance check:  
```c
warning: undefined symbol SOC_ESP32S3_PROCPU:

- Referenced at boards/espressif/esp32s3_devkitm/Kconfig.esp32s3_devkitm:4:

config BOARD_ESP32S3_DEVKITM
	bool
	select SOC_ESP32S3_MINI_N8
	select SOC_ESP32S3_PROCPU if BOARD_ESP32S3_DEVKITM_ESP32S3_PROCPU
	select SOC_ESP32S3_APPCPU if BOARD_ESP32S3_DEVKITM_ESP32S3_APPCPU
warning: undefined symbol SOC_ESP32S3_APPCPU:

```

Reply:
They should be selected in the board's `Kconfig` file not `Kconfig.<board>` file, also the `bool` line vanishes

Reply:
OK, moving into Kconfig.defconfig in its proper board if solved it.